wagtail/admin/templates/wagtailadmin/shared/dialog/dialog.html

```diff
@@ -0,0 +1,36 @@
+{% load wagtailadmin_tags i18n %}
+<div
+    id="{{ title|cautious_slugify }}"
+    aria-labelledby="title-{{ title|cautious_slugify }}"
+    aria-hidden="true"
+    data-dialog
+    class="w-fixed w-inset-0 w-z-dialog w-flex aria-hidden:w-hidden w-p-4"
```

Comment:
will there be any way to add custom classes to the dialog?

Reply:
Should this be a `template` tag instead? and when the modal opens it will put the template contents at the end of the DOM, otherwise there is a chance these dialogs will be added anywhere and adopt the styling of their container.

Reply:
Will there be any way to add additional data attributes here (or attributes in general)?

Reply:
- **custom classes / attrs:** I spoke with @thibaudcolas about this but he wasn't keen on having these as an option at this point. I do think having abilities to pass custom classes is definitely useful, although i'm not sure how much customization we are aiming to allow on this component as well. For now i've made an option for `{{ class_name }}` at least and will take handle any lashings from Thibaud 😁   

- **template tag** - Awesome suggestion/improvement! I've made this change now 😁

Reply:
Great. Let's do attrs later if needed. 

I think classname is a minimum requirement for these things in my opinion. Easy to add and gives a lot of flexibility for custom usage. Plus low risk in terms of html injection issues.

Reply:
Agreed!